### ISSUE TITLE
should `MinerAPI.getAvailableBalance` be `view`?

### ISSUE THREAD

**Issue by hcheng826** [Zondax/filecoin-solidity#359]
Or this function is expected to change state?

**Comment by maciejwitowski** [comment]
These functions aren't expected to modify the state. They all past the `static_call = true` flag. Eg [here](https://github.com/Zondax/filecoin-solidity/blob/master/contracts/v0.8/MinerAPI.sol#L93) (the last argument).

**Comment by maciejwitowski** [comment]
Yeah, this is tricky. You can see the explanation in [this Slack thread](https://filecoinproject.slack.com/archives/C04AT6ZKJC9/p1679260685977999), especially 

> solc's mutability checker sees a "delegatecall" and decides the method can't be view, because a delegatecall might be able to change state.

and
> If you want to mark those methods view, you have to fool the mutability checker.

[Here](https://github.com/wadealexc/fevmate/blob/main/contracts/utils/CallNative.sol#L53) is the example of "fooling the mutability checker".

**Comment by rllola** [comment]
I would not recommend doing this hack in your code to fool the checker. But indeed we can't put view on functions because all the call use `DelegateCall` opcode.

And to answer your second question, simply getting a value cost gas. In the PR you can see a gas consumption table (https://github.com/Zondax/filecoin-solidity/pull/358#issuecomment-1473998338). The values are taken from running the contract in a local fvm.
